src/sqlfluff/core/linter/common.py

```diff
@@ -0,0 +1,67 @@
+"""Defines small container classes to hold intermediate results during linting."""
+
+from typing import (
+    Any,
+    List,
+    NamedTuple,
+    Optional,
+    Tuple,
+)
+
+from sqlfluff.core.errors import SQLBaseError
+from sqlfluff.core.templaters import TemplatedFile
+from sqlfluff.core.config import FluffConfig
+from sqlfluff.core.parser.segments.base import BaseSegment
+
+
+class RuleTuple(NamedTuple):
+    """Rule Tuple object for describing rules."""
+
+    code: str
+    description: str
+
+
+class NoQaDirective(NamedTuple):
+    """Parsed version of a 'noqa' comment."""
+
+    line_no: int  # Source line number
+    rules: Optional[Tuple[str, ...]]  # Affected rule names
+    action: Optional[str]  # "enable", "disable", or "None"
+
+
+class ProtoFile(NamedTuple):

```

Comment:
What does "proto" mean here? It's almost a file but not a file just yet? 🤔

Reply:
Great question - I _think_ so, but I might try and come up with a better name for this. I'm going to need to rework some of this anyway. 🤔

Reply:
Turns out that this class is never referred to! Must be a chunk of Legacy code. I'm removing it.